worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/FileWriter.java

```diff
@@ -434,6 +431,21 @@ public PartitionSplitMode getSplitMode() {
     return splitMode;
   }
 
+  @Override
+  public void notifyError(String mountPoint, DiskStatus diskStatus) {
+    IOException ioe = new IOException("Device ERROR! Disk: " + mountPoint + " : " + diskStatus);
+    if (diskStatus == DiskStatus.READ_OR_WRITE_FAILURE) {
+      logger.error(
+          "Will destroy FileWriter {}, because disk {} status {}", this, mountPoint, diskStatus);
+      destroy(ioe);
+    } else {
+      if (!notifier.hasException()) {
+        notifier.setException(ioe);
+      }
+      deviceMonitor.unregisterFileWriter(this);
+    }
+  }
```

Comment:
Here we don't need to check `diskStatus == DiskStatus.READ_OR_WRITE_FAILURE`, when notify error, it means device down